File: Structural-Patterns/Decorator/C#/Decorator/Implementations/ShieldDecorator.cs
public class ShieldDecorator : CharacterDecorator {

    public ShieldDecorator(ICharacter character) : base(character) {}
    
    public override int GetDefense(){
        return base.GetDefense() + 20;
    }

    public override String GetDescription(){
        return base.GetDescription() + ", carrying a sturdy shield";
    }

    public override List<String> GetAbilities(){
        List<String> abilities = base.GetAbilities();
        abilities.Add("Block attack");
        return abilities;
    }
}

File: Structural-Patterns/Decorator/Java/Implementations/ShieldDecorator.java
package Implementations;

import java.util.List;

import Interfaces.CharacterDecorator;
import Interfaces.ICharacter;

public class ShieldDecorator extends CharacterDecorator {

    public ShieldDecorator(ICharacter character) {
        super(character);
    }
    
    @Override
    public int getDefense(){
        return super.getDefense() + 20;
    }

    @Override
    public String getDescription(){
        return super.getDescription() + ", carrying a sturdy shield";
    }

    @Override
    public List<String> getAbilities(){
        List<String> abilities = super.getAbilities();
        abilities.add("Block attack");
        return abilities;
    }
}
